src/arithmetic/arithmetic_expression_construct.c

```diff
@@ -411,31 +426,49 @@ static AR_ExpNode *_AR_ExpFromMapProjection(const cypher_astnode_t *expr) {
 			// { .name }
 			prop = cypher_ast_map_projection_property_get_prop_name(selector);
 			prop_name = cypher_ast_prop_name_get_value(prop);
-			children[i * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
+			children[j * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
 			AR_ExpNode *entity = AR_EXP_NewVariableOperandNode(entity_name);
-			children[i * 2 + 1] = AR_EXP_NewAttributeAccessNode(entity, prop_name);
+			children[j * 2 + 1] = AR_EXP_NewAttributeAccessNode(entity, prop_name);
+			j++;
 		} else if(t == CYPHER_AST_MAP_PROJECTION_LITERAL) {
 			// { v: n.v }
 			prop = cypher_ast_map_projection_literal_get_prop_name(selector);
 			prop_name = cypher_ast_prop_name_get_value(prop);
 			const cypher_astnode_t *literal_exp =
 				cypher_ast_map_projection_literal_get_expression(selector);
-			children[i * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
-			children[i * 2 + 1] = AR_EXP_FromASTNode(literal_exp);
+			children[j * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
+			children[j * 2 + 1] = AR_EXP_FromASTNode(literal_exp);
+			j++;
 		} else if(t == CYPHER_AST_MAP_PROJECTION_IDENTIFIER) {
 			// { v }
 			prop = cypher_ast_map_projection_identifier_get_identifier(selector);
 			prop_name = cypher_ast_identifier_get_name(prop);
-
-			children[i * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
-
-			children[i * 2 + 1] = AR_EXP_NewVariableOperandNode(prop_name);
+			children[j * 2] = AR_EXP_NewConstOperandNode(SI_ConstStringVal((char *)prop_name));
+			children[j * 2 + 1] = AR_EXP_NewVariableOperandNode(prop_name);
+			j++;
+		} else if(t == CYPHER_AST_MAP_PROJECTION_ALL_PROPERTIES) {
```

Comment:
`} else if(t == CYPHER_AST_MAP_PROJECTION_ALL_PROPERTIES) {`
->
`} else if(t == CYPHER_AST_MAP_PROJECTION_ALL_PROPERTIES && propertiesOp == NULL) {`

Otherwise we'll be leaking

Reply:
Thanks. Done. [8814e48](https://github.com/RedisGraph/RedisGraph/pull/2698/commits/8814e4827683d32aca947ccbe1fac3817267a3d8)

Reply:
To avoid "Unexpected AST node type", I changed the comparison as:
```
} else if(t == CYPHER_AST_MAP_PROJECTION_ALL_PROPERTIES) {
			// { .* }
			// Use properties() to get a map with all properties
			if(propertiesOp == NULL) {

```